cli/azd/pkg/infra/azure_resource_manager.go

```diff
@@ -64,13 +71,17 @@ func (rm *AzureResourceManager) GetDeploymentResourceOperations(ctx context.Cont
 	// Recursively append any resource group deployments that are found
 	for _, operation := range subOperations {
 		if operation.Properties.TargetResource.ResourceType == string(AzureResourceTypeDeployment) {
-			err = rm.appendDeploymentResourcesRecursive(ctx, subscriptionId, resourceGroupName, operation.Properties.TargetResource.ResourceName, &resourceOperations)
+			err = rm.appendDeploymentResourcesRecursive(ctx, scope.SubscriptionId(), resourceGroupName, operation.Properties.TargetResource.ResourceName, &resourceOperations)
 			if err != nil {
 				return nil, fmt.Errorf("appending deployment resources: %w", err)
 			}
 		}
 	}
 
+	if len(resourceOperations) == 0 {
+		resourceOperations = subOperations
+	}
```

Comment:
I think this is similarly unclear why it exists. I suspect it is related to handling resource group scopes vs. subscription scopes and I wonder if we should make it part of an explicit scope-type handling